compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -140,6 +147,19 @@ impl<'f> PerFunctionContext<'f> {
             let references = self.find_starting_references(block);
             self.analyze_block(block, references);
         }
+
+        // If we never load from an address within a function we can remove all stores to that address.
+        // This rule does not apply to reference parameters, which we must also check for before removing these stores.
+        for (block_id, block) in self.blocks.iter() {
+            let block_params = self.inserter.function.dfg.block_parameters(*block_id);
+            for (value, store_instruction) in block.last_stores.iter() {
+                let is_reference_param = block_params.contains(value)
+                    && self.inserter.function.dfg.value_is_reference(*value);
```

Comment:
```suggestion
```
You can get rid of this line since if we're loading from it it should be a reference